dpdispatcher/machine.py

```diff
@@ -73,16 +73,32 @@ def load_from_json(cls, json_path):
         return machine
 
     @classmethod
-    def load_from_dict(cls, machine_dict):
+    def dargs_check(cls, machine_dict={}, if_gen_docs=False):
+        doc_batch_type = 'The batch job system type. Option: Slurm, PBS, LSF, Shell, DpCloudServer'
+        doc_context_type = 'The connection used to remote machine. Option: LocalContext, LazyLocalContext, SSHContext， DpCloudServerContext'
+        doc_local_root = 'The dir where the tasks and relating files locate. Typically the project dir.'
+        doc_remote_root = 'The dir where the tasks are executed on the remote machine.'
+        doc_remote_profile = 'The information used to maintain the connection with remote machine.'
+
         machine_args = [
-            Argument("batch_type", str, optional=False),
-            Argument("context_type", str, optional=False),
-            Argument("local_root", str, optional=False),
-            Argument("remote_root", str, optional=True),
-            Argument("remote_profile", dict, optional=True),
+            Argument("batch_type", str, optional=False, doc=doc_batch_type),
+            Argument("context_type", str, optional=False, doc=doc_context_type),
+            Argument("local_root", str, optional=False, doc=doc_local_root),
+            Argument("remote_root", str, optional=True, doc=doc_remote_root),
+            Argument("remote_profile", dict, optional=True, doc=doc_remote_profile),
         ]
         machine_format = Argument("machine_dict", dict, machine_args)
-        machine_format.check_value(machine_dict)
+        if if_gen_docs:
+            ptr = machine_format.gen_doc()
+            return ptr
+        else:
+            machine_format.check_value(machine_dict)
+            return True
```

Comment:
This can be EXTREMELY confusing when you return two different types of value depending on the flag in the function parameters. I would suggest the following:
```python
@staticmethod # cls is not used so this should be static
def arginfo(): # I changed the function name to fit the return value
    # now define your arguments and docs here
    # ....
    machine_format = Argument("machine", dict, machine_args)
    return machine_format
```
Now whenever one needs to do arg check it can just call `Machine.arginfo().check_value(argdict)`
and doc generation would be like `Machine.arginfo().gen_doc()`.
Also, the returned Argument instance can be further composed to build larger arg trees. (for example, the remote_profile can be defined under SSH_Session class and passed here)